doc/source/developing/testing.rst

```diff
@@ -3,503 +3,400 @@
 How to Run the Unit Tests
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
-One can run the unit tests very straightforwardly from any python interpreter
-that can import the yt module:
-
-.. code-block:: python
-
-   import yt
-
-   yt.run_nose()
-
-If you are developing new functionality, it is sometimes more convenient to use
-the Nose command line interface, ``nosetests``. You can run the unit tests
-using ``nose`` by navigating to the base directory of the yt git
-repository and invoking ``nosetests``:
+One can run the unit tests by navigating to the base directory of the yt git
+repository and invoking ``pytest``:
 
 .. code-block:: bash
 
    $ cd $YT_GIT
-   $ nosetests
+   $ pytest
 
 where ``$YT_GIT`` is the path to the root of the yt git repository.
 
-If you want to specify a specific unit test to run (and not run the entire
-suite), you can do so by specifying the path of the test relative to the
-``$YT_GIT/yt`` directory -- note that you strip off one yt more than you
-normally would!  For example, if you want to run the plot_window tests, you'd
+If you only want to run tests in a specific file, you can do so by specifying the path of the test relative to the
+``$YT_GIT/`` directory. For example, if you want to run the ``plot_window`` tests, you'd
 run:
 
 .. code-block:: bash
 
-   $ nosetests yt/visualization/tests/test_plotwindow.py
+   $ pytest yt/visualization/tests/test_plotwindow.py
 
-Handling yt dependencies
-^^^^^^^^^^^^^^^^^^^^^^^^
+Additionally, if you only want to run, say, ``test_all_fields`` in ``plot_window.py``, you can do so by running:
 
-We attempt to make yt compatible with a wide variety of upstream software
-versions. However, sometimes a specific version of a project that yt depends on
-causes some breakage and must be blacklisted in the tests or a more
-experimental project that yt depends on optionally might change sufficiently
-that the yt community decides not to support an old version of that project.
+.. code-block:: bash
 
-To handle cases like this, the versions of upstream software projects installed
-on the machines running the yt test suite are pinned to specific version
-numbers that must be updated manually. This prevents breaking the yt tests when
-a new version of an upstream dependency is released and allows us to manage
-updates in upstream projects at our pace.
+    $ pytest yt/visualization/tests/test_plotwindow.py::test_all_fields
 
-If you would like to add a new dependency for yt (even an optional dependency)
-or would like to update a version of a yt dependency, you must edit the
-``tests/test_requirements.txt`` file, this path is relative to the root of the
-repository. This file contains an enumerated list of direct dependencies and
-pinned version numbers. For new dependencies, simply append the name of the new
-dependency to the end of the file, along with a pin to the latest version
-number of the package. To update a package's version, simply update the version
-number in the entry for that package.
```

Comment:
all of this information on dependencies is important for development. Has it moved elsewhere? We should have narrative documentation of it remaining in our development docs.

Reply:
now that you mention it... it's also a conflict with #3236, where I'm _updating_ this doc.